pytorch_lightning/utilities/auto_restart.py

```diff
@@ -480,6 +503,9 @@ def patch_dataloader_iterator(
     will extract the current iteration as part of the metadata returned by a custom batch.
     """
 
+    if not _FaultTolerantMode.detect_current_mode().is_automatic:
+        return
```

Comment:
Shouldn't this be checked wherever this gets called?

Reply:
Not sure to fully understand your question. Yes, it needs to be checked every time it gets called. But this code will be refactored soon.